core/capabilities/ccip/ccip_integration_tests/rmn/rmn_home_test.go

```diff
@@ -0,0 +1,162 @@
+
+	_, err = rmnHome.SetCandidate(uni.Transactor, staticConfig, dynamicConfig, zeroBytes)
+	require.NoError(t, err)
+	uni.Backend.Commit()
+
+	configDigest, err := rmnHome.GetCandidateDigest(&bind.CallOpts{})
+	require.NoError(t, err)
+
+	_, err = rmnHome.PromoteCandidateAndRevokeActive(uni.Transactor, configDigest, zeroBytes)
+	require.NoError(t, err)
+	uni.Backend.Commit()
+
+	rmnHomeBoundContract := types.BoundContract{
+		Address: rmnHomeAddress.String(),
+		Name:    consts.ContractNameRMNHome,
+	}
+
+	err = uni.HomeContractReader.Bind(testutils.Context(t), []types.BoundContract{rmnHomeBoundContract})
+	require.NoError(t, err)
+
+	rmnHomeReader := readerpkg.NewRMNHomePoller(uni.HomeContractReader, rmnHomeBoundContract, lggr, 1*time.Millisecond)
+
+	err = rmnHomeReader.Start(testutils.Context(t))
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		err1 := rmnHomeReader.Close()
+		require.NoError(t, err1)
+	})
+
+	//================================Test RMNHome Reader===============================
+	expectedNodesInfo := integrationhelpers.GenerateExpectedRMNHomeNodesInfo(staticConfig, chainID1)
+
+	testutils.AssertEventually(t, func() bool {
+		nodesInfo, err2 := rmnHomeReader.GetRMNNodesInfo(configDigest)
+		if err2 != nil {
+			t.Logf("Error getting RMN nodes info: %v", err2)
+			return false
+		}
+
+		if !assert.Equal(t, expectedNodesInfo, nodesInfo) {
+			t.Logf("Expected nodes info doesn't match actual nodes info")
+			return false
+		}
+
+		return true
+	})
+
+	// Add a new candidate config
+	staticConfig2, dynamicConfig2 := integrationhelpers.GenerateRMNHomeConfigs(
+		"PeerID2",
+		"DummyPublicKey2",
+		"This is a sample offchain configuration in the static config 2",
+		chainID2,
+		minObservers2,
+		big.NewInt(observerBitmap2),
+	)
+
+	_, err = rmnHome.SetCandidate(uni.Transactor, staticConfig2, dynamicConfig2, zeroBytes)
+	require.NoError(t, err)
+	uni.Backend.Commit()
+
+	candidateConfigDigest, err := rmnHome.GetCandidateDigest(&bind.CallOpts{})
+	require.NoError(t, err)
+
+	expectedCandidateNodesInfo := integrationhelpers.GenerateExpectedRMNHomeNodesInfo(staticConfig2, chainID2)
+
+	testutils.AssertEventually(t, func() bool {
+		nodesInfo, err2 := rmnHomeReader.GetRMNNodesInfo(candidateConfigDigest)
+		if err2 != nil {
+			t.Logf("Error getting RMN nodes info: %v", err2)
+			return false
+		}
+
+		if !assert.Equal(t, expectedCandidateNodesInfo, nodesInfo) {
+			t.Logf("Expected nodes info doesn't match actual nodes info")
+			return false
+		}
+
+		return true
+	})
+
+	// Promote the candidate config
+	_, err = rmnHome.PromoteCandidateAndRevokeActive(uni.Transactor, candidateConfigDigest, configDigest)
+	require.NoError(t, err)
+	uni.Backend.Commit()
+
+	testutils.AssertEventually(t, func() bool {
+		nodesInfo, err2 := rmnHomeReader.GetRMNNodesInfo(candidateConfigDigest)
+		if err2 != nil {
+			t.Logf("Error getting RMN nodes info: %v", err2)
+			return false
+		}
+
+		if !assert.Equal(t, expectedCandidateNodesInfo, nodesInfo) {
```

Comment:
slices.EqualFunc

Reply:
Since there's some duplication here you can extract the comparator func into a separate func